smart-contracts/contracts/cl-vault/src/msg.rs

```diff
@@ -20,8 +20,10 @@ pub enum ExtensionExecuteMsg {
     ModifyRange(ModifyRangeMsg),
     /// provides a fungify callback interface for the contract to use
     Merge(MergePositionMsg),
+    /// Collect any rewards from Osmosis to the Vault
+    CollectRewards {},
     /// Distribute any rewards over all users
-    DistributeRewards {},
+    DistributeRewards { amount_of_users: Uint128 },
```

Comment:
this works though I recommend using an offset/limit approach.

if we want to get this out fast this is not a major comment

Reply:
You mean like page and pageSize? We thought about that but then we thought also that this would need the executer (i.e. a cronjob) to keep track of the previous executed transaction. Like this we can rely on a unique argument like amount_of_users to process and if its enough to process them all we unlock, otherwise we keep it locked and we wait for another same-payload transaction.